src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -307,6 +310,28 @@ public function getRootDir()
     }
 
     /**
+     * Gets the application root dir (path of the project's composer file).
+     *
+     * @return string The project root dir
+     */
+    public function getProjectDir()
+    {
+        if (null === $this->projectDir) {
```

Comment:
This implementation is really strange. It basically assigns the variable twice. Once in the method and once in the constructor.